tokio/src/sync/mod.rs

```diff
@@ -431,6 +431,19 @@
 //!   number of permits, which tasks may request in order to enter a critical
 //!   section. Semaphores are useful for implementing limiting or bounding of
 //!   any kind.
+//!
+//! # Runtime compatibility
+//!
+//! All synchronization primitives provided in this module are runtime agnostic.
+//! You can freely move them between different instances of the Tokio runtime
+//! or even use them from non-Tokio runtimes.
+//!
+//! The only exception to this is the [`mpsc::Sender::send_timeout`] method.
```

Comment:
The only investigation I did on the matter of exceptions to runtime compatibility was to have a look at the feature gates inside the `sync` module. Based on this, the only relevant code I could find was `mpsc::Sender::send_timeout` gated by the `time` feature. Everything else looks runtime independent to me, unless there is something that relies on runtime-dependent functionality that is not feature-gated.

Reply:
It's probably worth mentioning that all `tokio::sync` utilities participate in [cooperative scheduling](https://docs.rs/tokio/latest/tokio/task/index.html#cooperative-scheduling) when used on a Tokio runtime. (But they still work outside of a Tokio runtime - this piece just does nothing in that case.)

Reply:
I've added a paragraph on cooperative scheduling. Out of interest, would you mind pointing me to an example where we have an explicit yield back to the cooperative scheduler?

Reply:
Sure. There's one here:

https://github.com/tokio-rs/tokio/blob/91169992b2ed0cf8844dbaa0b4024f9db588a629/tokio/src/sync/mpsc/chan.rs#L294-L295

Reply:
Thanks, I somehow missed `poll_proceed` when sifting through the `runtime::coop` module, leaving me a bit confused as to how one actually interacts with a `Budget`.